cpp/build-support/lint_cpp_cli.py

```diff
@@ -30,6 +30,7 @@
 _STRIP_COMMENT_REGEX = re.compile('(.+)?(?=//)')
 _NULLPTR_REGEX = re.compile(r'.*\bnullptr\b.*')
 _RETURN_NOT_OK_REGEX = re.compile(r'.*\sRETURN_NOT_OK.*')
+_RETURN_NOT_OK_REGEX = re.compile(r'.*\sASSIGN_OR_RAISE.*')
```

Comment:
```suggestion
_ASSIGN_OR_RAISE = re.compile(r'.*\sASSIGN_OR_RAISE.*')
```

Reply:
Fixed.  thanks.  also added a unit test for the public API.